### DISCUSSION TITLE
Problem with block parameter when defining custom ResNet model

### DISCUSSION THREAD

**Discussion by rajanish4** [Project-MONAI/MONAI#4678] [Q&A]
Hi there,
I am trying to define a custom ResNet model, however, for the block parameter, the documentation says it should either be basic or bottleneck, and these blocks cant be imported or written as string values to this keyword (error when providing string values). I would like to know what value is to be passed to this parameter. Please let me know. My code is as follows:
Network = ResNet(block='basic', layers=[1, 1, 1, 1], block_inplanes=[8, 16, 32, 64],
                               spatial_dims=3, num_classes=2, n_input_channels=1).cuda()
I believe the ResNetBlock should be imported which is not possible currently.

[python.txt](https://github.com/Project-MONAI/MONAI/files/9091724/python.txt)

**Comment by wyli** [comment]
good question I think the string type support is somehow missing here. anyway, the current usage is:
```py
from monai.networks.nets import ResNet                                                                                                                                                                                   
from monai.networks.nets.resnet import ResNetBlock                                                                                                                                                                                                                            
                                                                                                                                                                                                                                                                                
network = ResNet(block=ResNetBlock, layers=[1, 1, 1, 1], block_inplanes=[8, 16, 32, 64],                                                                                                                                                                                      
                 spatial_dims=3, num_classes=2, n_input_channels=1).cuda()                                                                                                                                                                                                    
print(network) 
```

(created a feature request https://github.com/Project-MONAI/MONAI/issues/4691)

**Reply by rajanish4** [reply]
Thanks, thats right, i did it that way, but for that to work, the __init__.py should be modified here to include ResNetBlock and ResNetBottleneck here: monai/networks/nets/__init__.py. Currently it is not possible to import those blocks.

> good question I think the string type support is somehow missing here. anyway, the current usage is:
> 
> ```python
> from monai.networks.nets import ResNet                                                                                                                                                                                   
> from monai.networks.nets.resnet import ResNetBlock                                                                                                                                                                                                                            
>                                                                                                                                                                                                                                                                                 
> network = ResNet(block=ResNetBlock, layers=[1, 1, 1, 1], block_inplanes=[8, 16, 32, 64],                                                                                                                                                                                      
>                  spatial_dims=3, num_classes=2, n_input_channels=1).cuda()                                                                                                                                                                                                    
> print(network) 
> ```
> 
> (created a feature request #4691)
